modules/indexer/code/indexer.go

```diff
@@ -188,11 +132,7 @@ func Init() {
 					code.gitea.io/gitea/modules/indexer/code.index(indexer.go:105)
 				*/
 				if err := index(ctx, indexer, indexerData.RepoID); err != nil {
-					if !idx.Ping() {
-						log.Error("Code indexer handler: indexer is unavailable.")
-						unhandled = append(unhandled, indexerData)
-						continue
-					}
+					unhandled = append(unhandled, indexerData)
```

Comment:
There are two cases.

#### 1. The data is invalid and the engine cannot index it forever.

The old code try to ignore the data (but it doesn't work very well), but I don't think it's right. Users may report "I cannot see the issue/code in the search results even though it does exist", and they coundn't find anything in the logs because the indexing error may happened weeks or even years ago.

The new code will requeue the invalid data and retry to index them, and get failed again and again, so the users could notice the error logs and report it. I hope so, because it's a serious bug which has to be fixed. So we can: fix the bug, upgrade index version, ask the user to upgrade Gitea. They don't need to empty the queue (the invalid data will be valid in new version), or delete the index (Gitea will build a new index)

Don't worry about crazy retrying, the queue will backoff: https://github.com/go-gitea/gitea/blob/18f26cfbf7f9b36b838c0e8762bfba98c89b9797/modules/queue/workergroup.go#L102

#### 2. The engine is unavailable momentarily, and the data can be indexed then it's avaliable.

Of cause Gitea should retry again and again, the engine coule be down for days, and there's no reason to give up and make search results inaccurate.